API/Controllers/UserController.cs

```diff
@@ -166,5 +172,44 @@ public async Task<IActionResult> DeleteAccount(int userId)
             userService.Save();
             return Ok();
         }
+
+        /// <summary>
+        ///     Delete all user data
+        /// </summary>
+        /// <returns></returns>
+        [HttpDelete("{userId}/allData")]
+        [Authorize(Policy = nameof(Defaults.Scopes.UserWrite))]
+        public async Task<IActionResult> DeleteAllUserData(int userId)
+        {
+            if(await userService.FindAsync(userId) == null)
+            {
+                ProblemDetails problem = new ProblemDetails
+                {
+                    Title = "Failed getting the user account.",
+                    Detail = "The database does not contain a user with this student id.",
+                    Instance = "TODO-CHANGE-TO-GENERATED-INSTANCE-CODE"
+                };
+                return NotFound(problem);
+            }
+            List<Project> projects = (List<Project>) await projectService.GetAllWithUsersAsync();
```

Comment:
```suggestion
            List<Project> projects = await projectService.GetAllWithUsersAsync();
```
Why cast it when it already returns a list of projects?